ares.h

```diff
@@ -368,8 +368,8 @@ struct iovec;
 struct ares_socket_functions {
    ares_socket_t(*asocket)(int, int, int, void *);
    int(*aclose)(ares_socket_t, void *);
-   int(*aconnect)(ares_socket_t, const struct sockaddr *, socklen_t, void *);
-   ssize_t(*arecvfrom)(ares_socket_t, void *, size_t, int, struct sockaddr *, socklen_t *, void *);
+   int(*aconnect)(ares_socket_t, const struct sockaddr *, ares_socklen_t, void *);
+   ssize_t(*arecvfrom)(ares_socket_t, void *, size_t, int, struct sockaddr *, ares_socklen_t *, void *);
    ssize_t(*asendv)(ares_socket_t, const struct iovec *, int, void *);
```

Comment:
`ares_setup.h` is not a public header so we can't use that for actual applications or external library users. I suggest we add magic that adds a `ares_ssize_t` type to `ares_build.h` much in the same style `ares_socklen_t` is defined.